### PR TITLE
Fix route viewer fitting pt2

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
Requires/blocked by https://github.com/opentripplanner/otp-ui/pull/507.

This PR optimizes rendering in the route viewer by preventing the stops overlay from displaying all transit stops when selecting one route after the other in the route viewer.

### CODE REVIEW

File: lib/components/map/connected-stops-overlay.js [hunk 1]
```diff
@@ -1,64 +1,62 @@
...
-          prev.push(cur)
-        }
-        return prev
-      }, [])
-      // Override the minimum zoom so that the stops appear even if zoomed out
-      minZoom = 2
-    }
+      return prev
+    }, [])
+    minZoom = 2
```
Comment (miles-grant-ibigroup) [inline lib/components/map/connected-stops-overlay.js:43]: The comment about this got lost!

**Review by miles-grant-ibigroup — APPROVED** [review]
Will wait for the upcoming behavior change to fully test functionality but looks a lot cleaner for now! Some comments seem to have gotten lost but I suspect there will be large changes here soon anyway.

File: lib/components/map/connected-stops-overlay.js (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/map/connected-stops-overlay.js:44]: Oops, I added the comment back in https://github.com/opentripplanner/otp-react-redux/pull/720/commits/6e6a9726464f4793b915a7c01c1bcd3c7c1d44cc.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
This works well for me and there's not noticeable delay, so happy to see it!  Thank you!

**Pull request merged**